src/soc/ibm/power9/istep_14_5.c

```diff
@@ -257,81 +294,63 @@ static void fir_unmask(int mcs_i)
- *
- * a) p9_mss_setup_bars.C (proc chip) -- Nimbus
- * b) p9c_mss_setup_bars.C (proc chip) -- Cumulus
- *    - Same HWP interface for both Nimbus and Cumulus, input target is
- *      TARGET_TYPE_PROC_CHIP; HWP is to figure out if target is a Nimbus (MCS)
- *      or Cumulus (MI) internally.
- *    - Prior to setting the memory bars on each processor chip, this procedure
- *      needs to set the centaur security protection bit
- *      - TCM_CHIP_PROTECTION_EN_DC is SCOM Addr 0x03030000
- *      - TCN_CHIP_PROTECTION_EN_DC is SCOM Addr 0x02030000
- *      - Both must be set to protect Nest and Mem domains
- *    - Based on system memory map
- *      - Each MCS has its mirroring and non mirrored BARs
- *      - Set the correct checkerboard configs. Note that chip flushes to
- *        checkerboard
- *      - need to disable memory bar on slave otherwise base flush values will
- *        ack all memory accesses
- * c) p9_setup_bars.C
- *    - Sets up Powerbus/MCD, L3 BARs on running core
- *      - Other cores are setup via winkle images
- *    - Setup dSMP and PCIe Bars
- *      - Setup PCIe outbound BARS (doing stores/loads from host core)
- *        - Addresses that PCIE responds to on powerbus (PCI init 1-7)
- *      - Informing PCIe of the memory map (inbound)
- *        - PCI Init 8-15
- *    - Set up Powerbus Epsilon settings
- *      - Code is still running out of L3 cache
- *      - Use this procedure to setup runtime epsilon values
- *      - Must be done before memory is viable
- */
-void istep_14_5(void)
+static void proc_setup_bars(uint8_t chip)
 {
 	int mcs_i;
-	printk(BIOS_EMERG, "starting istep 14.5\n");
-	report_istep(14,5);
 
 	/* Start MCS reset */
-	revert_mc_hb_dcbz_config();
+	revert_mc_hb_dcbz_config(chip);
 
-	fill_groups();
+	fill_groups(chip);
 
 	for (mcs_i = 0; mcs_i < MCS_PER_PROC; mcs_i++) {
 		chiplet_id_t nest = mcs_to_nest[mcs_ids[mcs_i]];
 
-		if (!mem_data.mcs[mcs_i].functional)
+		if (!mem_data[chip].mcs[mcs_i].functional)
 			continue;
 
-		fir_unmask(mcs_i);
+		fir_unmask(chip, mcs_i);
 
 		/*
 		 * More undocumented registers. First two are described before
 		 * 'mcfgp_regs', last two are for setting up memory hole and SMF, they
 		 * are unused now.
 		 */
-		write_scom_for_chiplet(nest, 0x0501080A, mcfgp_regs[mcs_i][0]);
-		write_scom_for_chiplet(nest, 0x0501080C, mcfgp_regs[mcs_i][1]);
-		write_scom_for_chiplet(nest, 0x0501080B, 0);
-		write_scom_for_chiplet(nest, 0x0501080D, 0);
+		write_rscom_for_chiplet(chip, nest, 0x0501080A, mcfgp_regs[mcs_i][0]);
+		write_rscom_for_chiplet(chip, nest, 0x0501080C, mcfgp_regs[mcs_i][1]);
+		write_rscom_for_chiplet(chip, nest, 0x0501080B, 0);
+		write_rscom_for_chiplet(chip, nest, 0x0501080D, 0);
 	}
 
-	mcd_fir_mask();
+	mcd_fir_mask(chip);
+}
+
+void istep_14_5(uint8_t chips)
+{
+	uint8_t chip;
+
+	printk(BIOS_EMERG, "starting istep 14.5\n");
+	report_istep(14,5);
+
+	/* Start MCS reset */
+	revert_mc_hb_dcbz_config(/*chip=*/0);
```

Comment:
This is called in `proc_setup_bars()`, why also here?

Reply:
I meant to move it from `proc_setup_bars()`, HB calls it only for the master CPU.

Reply:
Makes sense, other CPUs shouldn't have their cache initialized by SBE.